main.py

```diff
@@ -104,42 +154,80 @@
 			Sy_word.append(line.rstrip("\n"))
 	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config)
 
+	if postprocess_words:
+		log_file="log_pipeline_postprocess.csv"
+	else:
+		if finetune_embedding:
+			log_file="log_pieline_finetune.csv"
+		else:
+			log_file="log_pipeline.csv"
+	
 	# Initialize final model
-	model = Model(config=config,pipeline=True)
+	model = Model(config=config,pipeline=True,finetune=finetune_embedding)
 
 	# Train the final model
 	trainer = Trainer(model=model, config=config)
 	if restart: trainer.load_checkpoint()
 
 	for epoch in range(config.training_num_epochs):
 		print("========= Epoch %d of %d =========" % (epoch+1, config.training_num_epochs))
-		train_intent_acc, train_intent_loss = trainer.pipeline_train_decoder(train_dataset, postprocess_words,log_file="log_pipeline_postprocess.csv")
-		valid_intent_acc, valid_intent_loss = trainer.pipeline_test_decoder(valid_dataset,  postprocess_words,log_file="log_pipeline_postprocess.csv")
+		train_intent_acc, train_intent_loss = trainer.pipeline_train_decoder(train_dataset, postprocess_words,log_file=log_file)
+		valid_intent_acc, valid_intent_loss = trainer.pipeline_test_decoder(valid_dataset,  postprocess_words,log_file=log_file)
 
 		print("========= Results: epoch %d of %d =========" % (epoch+1, config.training_num_epochs))
 		print("*intents*| train accuracy: %.2f| train loss: %.2f| valid accuracy: %.2f| valid loss: %.2f\n" % (train_intent_acc, train_intent_loss, valid_intent_acc, valid_intent_loss) )
+		if postprocess_words:
+			trainer.save_checkpoint(model_path="model_state_postprocess.pth")
+		else:
+			if finetune_embedding:
+				trainer.save_checkpoint(model_path="model_state_pipeline_finetune.pth")
+			else:
+				trainer.save_checkpoint(model_path="model_state_pipeline.pth")
 
-		trainer.save_checkpoint(model_path="model_state_postprocess.pth")
-
-	test_intent_acc, test_intent_loss = trainer.pipeline_test_decoder(test_dataset, postprocess_words,log_file="log_pipeline_postprocess.csv")
+	test_intent_acc, test_intent_loss = trainer.pipeline_test_decoder(test_dataset, postprocess_words,log_file=log_file)
 	print("========= Test results =========")
 	print("*intents*| test accuracy: %.2f| test loss: %.2f| valid accuracy: %.2f| valid loss: %.2f\n" % (test_intent_acc, test_intent_loss, valid_intent_acc, valid_intent_loss) )
 
 if pipeline_gold_train:
 	# Generate datasets
-	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config,use_gold_utterances=True)
+	train_dataset, valid_dataset, test_dataset = get_SLU_datasets(config,use_gold_utterances=True,random_split=random_split, disjoint_split=disjoint_split)
 
 	# Initialize final model
-	model = Model(config=config,pipeline=True)
+	if use_semantic_embeddings:
+		glove_embeddings=obtain_glove_embeddings(semantic_embeddings_path, train_dataset.Sy_word )
```

Comment:
no fasttext option here?

Reply:
Since performance in end to end much greater than performance in pipeline, I decided to not further experiment with pipeline manner. Hence did not added fasttext option here.